core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -562,8 +562,9 @@ public void refresh(String source) throws EngineException {
             ensureOpen();
             searcherManager.maybeRefreshBlocking();
         } catch (AlreadyClosedException e) {
-            ensureOpen();
```

Comment:
this is the code that I mean, is it ok to concurrently close SM and  call SM#maybeRefreshBlocking? if so we can ignore the ACE if ensureOpen throws an exception so I thin the code is good?